crates/storage/provider/src/providers/database/mod.rs

```diff
@@ -467,11 +467,19 @@ impl<N: ProviderNodeTypes> TransactionsProvider for ProviderFactory<N> {
         &self,
         range: impl RangeBounds<TxNumber>,
     ) -> ProviderResult<Vec<Address>> {
-        self.provider()?.senders_by_tx_range(range)
+        if EitherWriter::senders_destination(self).is_static_file() {
+            self.static_file_provider.senders_by_tx_range(range)
+        } else {
+            self.provider()?.senders_by_tx_range(range)
+        }
```

Comment:
just a note, that if we use the db tool to change the destination, we lose access to the previous senders. just noting and its fine by me, its not something users should be using anyway